app/questionnaire/questionnaire_schema.py

```diff
@@ -135,17 +140,25 @@ def get_parent_options_for_block(self, block_id):
                 options_with_children.update(answer_options_with_children)
         return options_with_children
 
-    def group_has_dependent_repeat_groups(self, group_id):
-        """ Given a group_id return True if any other group repeats over this group
-            in a `repeat_rule.[].repeat.groups`.
-            Otherwise return False (including for non-existant group_id arguments)"""
-        return bool(self._groups_used_in_repeating_groups[group_id])
-
-    def group_has_group_repeat_rule(self, group_id):
-        """ Returns True if a group has group dependencies on it's repeat rule, otherwise False"""
-        rule = self.get_repeat_rule(self.get_group(group_id))
-        if rule and rule.get('group_ids'):
-            return True
+    def get_group_dependencies(self, group_id):
+        return self.group_dependencies.get(group_id)
+
+    def get_group_dependencies_group_drivers(self):
+        return self.group_dependencies['group_drivers']
+
+    def get_group_dependencies_block_drivers(self):
+        return self.group_dependencies['block_drivers']
+
+    def location_requires_group_instance(self, location):
+        return bool(self.group_dependencies.get(location.group_id) or
+                    location.group_id in self.get_group_dependencies_group_drivers() or
+                    location.block_id in self.get_group_dependencies_block_drivers())
+
+    def is_block_relationship_type(self, block_id):
```

Comment:
Maybe we can make this more generic, eg `is_block_of_type(self, block_id, type)`
otherwise we can end up with lots of functions

Reply:
This might be a bit misleading. This checks that the questions within the block are of type relationship. Not the block['type'] itself.
I just hated this `if location.block_id in ['relationships', 'household-relationships']:` to signify that it's a relationship block.

Maybe change to `has_block_question_type(self, block_id, type)`.

Reply:
`block_has_question_type(self, block_id, question_type)`?